modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -162,42 +504,797 @@ public void testListWithGetInt(String filter, int startIndexInt, int countInt,
+    public Object[][] dataToTestCreateProvidedUserManagerHandlerIsNull()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+
+        return new Object[][]{
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", null, user, ResponseCodeConstants.CODE_INTERNAL_ERROR}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateProvidedUserManagerHandlerIsNull")
+    public void testCreateProvidedUserManagerHandlerIsNull(String scimObjectString, String attributes,
+                                                           String excludeAttributes, Object objectUser,
+                                                           int expectedScimResponseStatus)
+            throws ConflictException, BadRequestException, CharonException {
+
+        User user = (User) objectUser;
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new InternalErrorException()));
+        Mockito.when(userManager.createUser(any(User.class), anyMap())).thenReturn(user);
+
+        SCIMResponse scimResponse = userResourceManager.create(scimObjectString,
+                null, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForTestCreatedUserResourceIsNull")
+    public Object[][] dataToTestCreatedUserResourceIsNull() {
+
+        return new Object[][]{
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", null, ResponseCodeConstants.CODE_INTERNAL_ERROR}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreatedUserResourceIsNull")
+    public void testCreatedUserResourceIsNull(String scimObjectString, String attributes,
+                                              String excludeAttributes, int expectedScimResponseStatus)
+            throws ConflictException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new InternalErrorException()));
+        Mockito.when(userManager.createUser(any(User.class), anyMap())).thenReturn(null);
+
+        SCIMResponse scimResponse = userResourceManager.create(scimObjectString,
+                userManager, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+
```

Comment:
Unnecessary new line

Reply:
Fixed in commit 6f5ca461fa07ac4b49182320325156a58b80b910